### PR TITLE
feat: add nested authentication input data

### LINKED ISSUE #9: Add nested input data example
__Is your feature request related to a problem? Please describe.__

The template does not depict how to do nested input data objects.

__Describe the solution you'd like__

I would like to see something like an authentication object in the input data that consists of a user name and token or something like that.

__Describe alternatives you've considered__

Not contain nested objects in the template.

__Additional context__

n/a

### PR DESCRIPTION
**Pull request by tmetzke**
## Description

closes #9

### CODE REVIEW

File: src/main/java/io/camunda/connector/Authentication.java [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+  }
+
+  @Override
+  public int hashCode() {
+    return Objects.hash(token, user);
+  }
+
+  @Override
+  public boolean equals(Object obj) {
+    if (this == obj) return true;
```
Comment (igpetrov) [inline src/main/java/io/camunda/connector/Authentication.java:50]: It's strange that auto-formatter doesn't catch such things. According to the Google Style Guide [braces are always used](https://google.github.io/styleguide/javaguide.html#s4.1.1-braces-always-used)

```suggestion
    if (this == obj) {
      return true;
    }
```

File: src/main/java/io/camunda/connector/Authentication.java [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+    if (obj == null) return false;
```
Comment (igpetrov) [inline src/main/java/io/camunda/connector/Authentication.java:51]:
```suggestion
    if (obj == null) {
      return false;
    }
```

File: src/main/java/io/camunda/connector/Authentication.java [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+    if (getClass() != obj.getClass()) return false;
```
Comment (igpetrov) [inline src/main/java/io/camunda/connector/Authentication.java:52]:
```suggestion
    if (getClass() != obj.getClass()) {
      return false;
    }
```

File: src/main/java/io/camunda/connector/MyConnectorRequest.java [hunk 4]
```diff
@@ -51,12 +49,12 @@ public boolean equals(Object obj) {
     if (obj == null) return false;
```
Comment (igpetrov) [inline src/main/java/io/camunda/connector/MyConnectorRequest.java:49]:
```suggestion
    if (obj == null) {
      return false;
    }
```

File: src/main/java/io/camunda/connector/MyConnectorRequest.java [hunk 5]
```diff
@@ -51,12 +49,12 @@ public boolean equals(Object obj) {
...
     if (getClass() != obj.getClass()) return false;
```
Comment (igpetrov) [inline src/main/java/io/camunda/connector/MyConnectorRequest.java:50]:
```suggestion
    if (getClass() != obj.getClass()) {
      return false;
    }
```

**Review by nikku — APPROVED** [review]
I think that should make some peoples lifes easier :clap:

File: element-templates/template-connector.json [hunk 6]
```diff
@@ -41,13 +41,28 @@
         }
       },
       {
```
Comment (igpetrov) [inline element-templates/template-connector.json:43]: Suggestion: I am wondering, whether template should also contain one dynamic variable example?

**Review by igpetrov — APPROVED** [review]
LGTM, thank you.

**Comment by nikku** [comment]
@igpetrov My bad for merging this early.

File: element-templates/template-connector.json (hunk 6 above)
Comment (nikku) [reply element-templates/template-connector.json:43]: We could add a separate template with optional authentication to showcase this.

**Comment by igpetrov** [comment]
> @igpetrov My bad for merging this early.

No worries, those comments were minor.

**Pull request merged**